### PR TITLE
Validation of GAU Record in NPSP Settings

### LINKED ISSUE #3020: Validation of GAU record needed in NPSP Settings
This has come up on the Hub a few times and in cases. If for whatever reason a GAU record that is referenced as the Default Allocation is deleted or changed, users are unable to create an Opportunity due to this error:

`Unable to create new Opportunities. npsp.TDTM_Opportunity: execution of AfterInsert caused by: System.DmlException: Insert failed. First exception on row 0; first error: INSUFFICIENT_ACCESS_ON_CROSS_REFERENCE_ENTITY, insufficient access rights on cross-reference id: a0....: [] (npsp)`

Where the reference ID refers to a GAU record.

This happens most frequently in new sandboxes.

Suggest that NPSP Settings check for record with ID referenced, and if it does not exist 'Default Allocation' is unchecked and value removed.

### PR DESCRIPTION
**Pull request by meena-sfdc**
# Critical Changes

# Changes

 - We introduced a change to Default Allocations Settings. When you delete the Default General Accounting Unit record, NPSP deselects the Use Default Allocation setting and clears the Default General Accounting Unit field when a user loads the NPSP Settings page.   

# Issues Closed
- Fixes #3020

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by meena-sfdc** [comment]
**lurch: attach W-023674

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AJfaeAAD" target="_blank">W-023674</a>

File: src/classes/STG_SettingsManager_CTRL.cls [hunk 1]
```diff
@@ -70,17 +70,20 @@ public with sharing class STG_SettingsManager_CTRL extends STG_Panel {
...
     public void onNpspSettingsLoad() {
+
         //Set TDTM config
         STG_InstallScript installScript = new STG_InstallScript();
         installScript.runNewOrgScript();
 
         UTIL_MasterSchedulableHelper.abortOldScheduledJobs();
 
-       if (STG_Panel.runningUserIsAdmin()) {
+        if (STG_Panel.runningUserIsAdmin()) {
```
[the remark this answers is not in the document]
Comment (meena-sfdc) [reply src/classes/STG_SettingsManager_CTRL.cls]: Its not aligned, so I added the extra space.

[the remark this answers is not in the document]
Comment (force2b) [reply src/classes/STG_SettingsManager_CTRL.cls]: @meena-sfdc Great catch.

**Pull request merged**